File: itx-java/itx-backend/src/main/java/io/github/chehsunliu/itx/backend/error/BackendException.java
package io.github.chehsunliu.itx.backend.error;

public class BackendException extends RuntimeException {
    private final int status;

    public BackendException(int status, String message) {
        super(message);
        this.status = status;
    }

    public int status() {
        return status;
    }

    public static BackendException notFound() {
        return new BackendException(404, "not found");
    }

    public static BackendException badRequest(String message) {
        return new BackendException(400, message);
    }

    public static BackendException unknown(String message) {
        return new BackendException(500, message);
    }
}

File: itx-kt/itx-backend/src/main/kotlin/io/github/chehsunliu/itx/backend/error/BackendException.kt
package io.github.chehsunliu.itx.backend.error

import io.ktor.http.HttpStatusCode

class BackendException(
    val status: HttpStatusCode,
    message: String,
) : RuntimeException(message) {
    companion object {
        fun notFound() = BackendException(HttpStatusCode.NotFound, "not found")

        fun badRequest(message: String) = BackendException(HttpStatusCode.BadRequest, message)

        fun unknown(message: String) = BackendException(HttpStatusCode.InternalServerError, message)
    }
}
